random_image_generator/RandomImageGenerator.py

```diff
@@ -1,193 +1,213 @@
+        text_xy = (top_left_corner_x, top_left_corner_y)
+        
+# output --> txt
+        with open('123.txt', 'wb') as f:
+            csv.writer(f, delimiter=' ').writerows((0,0))
+        if  text_size_x < top_left_corner_x & text_size_y<top_left_corner_y:
+            with open('123.txt', 'wb') as f:
+                csv.writer(f, delimiter=' ').writerows((text_size_x,text_size_y))
+        else:
+            with open('123.txt', 'wb') as f:
+                csv.writer(f, delimiter=' ').writerows(text_xy)
+            
+        # Random Color and Transparency (up to a reasonable range)
+        sample_color = rgba_image.getpixel((  # Tuple
+            int((top_left_corner_x + 0.5 * text_size_x) * size_factor),
+            int((top_left_corner_y + 0.5 * text_size_y) * size_factor)
+        ))
+        
+        # Background color matching check (TODO: SAMPLE MORE POINTS OR USE BOX MEAN)
+        while True:
+            red, green, blue = randint(0, 50), randint(0, 50), randint(0, 50)
+            transparency = randint(100, 255)
+            if abs(red - sample_color[0]) + abs(green - sample_color[1]) + abs(blue - sample_color[2]) > 100:
+                break
+            red, green, blue = 255 - red, 255 - green, 255 - blue
+            if abs(red - sample_color[0]) + abs(green - sample_color[1]) + abs(blue - sample_color[2]) > 100:
+                break
+        
+        # Overlay texts onto the image
+        image_draw.text(text_xy, text, font=font, fill=(red, green, blue, transparency))
+        text_overlay = text_overlay.resize(rgba_image.size)
+        image_with_text = Image.alpha_composite(rgba_image, text_overlay)
+
+        return image_with_text
+
+    def __add_watermark_to_image(self, image, watermark):
+        rgba_image = image.convert('RGBA')
+        rgba_watermark = watermark.convert('RGBA')
+  
+        image_x, image_y = rgba_image.size
+        watermark_x, watermark_y = rgba_watermark.size
+  
+        scale = randint(8, 12)
+        watermark_scale = max(image_x / (scale * watermark_x), image_y / (scale * watermark_y))
+        new_size = (int(watermark_x * watermark_scale), int(watermark_y * watermark_scale))
+        rgba_watermark = rgba_watermark.resize(new_size, resample=Image.ANTIALIAS)
+        rgba_watermark_mask = rgba_watermark.convert("L").point(lambda x: min(x, 180))
+        rgba_watermark.putalpha(rgba_watermark_mask)
+  
+        watermark_x, watermark_y = rgba_watermark.size
+
+        if random() > 0.5: # Random half
+            rgba_image.paste(rgba_watermark, (image_x - watermark_x, image_y - watermark_y), rgba_watermark_mask) # bot-right
+        else:
+            rgba_image.paste(rgba_watermark, (image_x - watermark_x, 0), rgba_watermark_mask) # top-right
+  
+        return rgba_image
+
+    def english_generator(self, place="random"):
+        if place not in ["random", "corner"]:
+            raise ValueError
+        for image in tqdm(self.images):

```

Comment:
The actual output should happen in these functions. It's better to change the return values of the above private functions to include the bounding boxes, and output them into txt files over here.